pymongo/asynchronous/mongo_client.py

```diff
@@ -1140,13 +1249,16 @@ def option_repr(option: str, value: Any) -> str:
             return f"{option}={value!r}"
 
         # Host first...
-        options = [
-            "host=%r"
-            % [
-                "%s:%d" % (host, port) if port is not None else host
-                for host, port in self._topology_settings.seeds
+        if hasattr(self, "_topology"):
+            options = [
+                "host=%r"
+                % [
+                    "%s:%d" % (host, port) if port is not None else host
+                    for host, port in self._topology_settings.seeds
+                ]
             ]
-        ]
+        else:
+            options = ["host={self._host}", "port={self._port}"]
```

Comment:
This is not safe. `self._host` is the entire URI so doing this would leak the user credentials.

Reply:
oh, i see, what should I use in the repr here instead then?

Reply:
What about this?:
```python
AsyncMongoClient(host='mongodb+srv://fqdn', ...
```